tokio-udp/src/socket.rs

```diff
@@ -52,13 +53,19 @@ impl UdpSocket {
         self.io.get_ref().connect(*addr)
     }
 
-    #[deprecated(since = "0.1.2", note = "use poll_send instead")]
-    #[doc(hidden)]
-    pub fn send(&mut self, buf: &[u8]) -> io::Result<usize> {
-        match self.poll_send(buf)? {
-            Async::Ready(n) => Ok(n),
-            Async::NotReady => Err(io::ErrorKind::WouldBlock.into()),
-        }
+    /// Returns a future that sends data on the socket to the remote address to which it is connected.
+    /// On success, the future will resolve to the number of bytes written.
+    ///
+    /// The [`connect`] method will connect this socket to a remote address. The future
+    /// will resolve to an error if the socket is not connected.
+    ///
+    /// [`connect`]: #method.connect
+    ///
+    /// # Panics
+    ///
+    /// If the future is polled outside of a task context.
+    pub fn send<'socket, 'buf>(&'socket mut self, buf: &'buf [u8]) -> Send<'socket, 'buf> {
```

Comment:
I believe you can elide the 'socket lifetime?

Reply:
Yes, it's possible, but mark it explicitly seems clearer.

Reply:
Not always true, generally speaking anytime you have a `fn(&self)` you know there is an anon lifetime so it doesnt make much sense to include it as it just adds noise.

Reply:
It seems very strange to elide only one lifetime. It breaks symmetry. And I don't see code out there do this:

* https://rust-lang-nursery.github.io/futures-api-docs/0.3.0-alpha.16/futures/io/trait.AsyncReadExt.html?search=#method.copy_into
* https://docs.rs/runtime/0.3.0-alpha.6/runtime/net/struct.UdpSocket.html#method.send_to
* https://docs.rs/romio/0.3.0-alpha.9/romio/udp/struct.UdpSocket.html#method.send_to